model_analyzer/result/run_config_result.py

```diff
@@ -43,15 +43,16 @@ def __init__(self, model_name, run_config, comparator, constraints=None):
             RunConfigResults and returns 1 if the first is better than
             the second, 0 if they are equal and -1
             otherwise
-        constraints: dict
+        constraints: dict (single model) or list of dicts (multi-model)
```

Comment:
We are probably relying on built-in types (dict/list) too much. I don't like having different types coming in to a function. Really seems like we need some kind of constraint class that hides a lot of implementation details. Likely not part of this story though.

Reply:
Yeah, I agree this is awkward

Reply:
TMA-732 has been filed to track this issue